<launch>

    <arg name="car" default="pegasus_v2"/>

    <group if="$(eval arg('car') == 'pegasus_v1')">

        <node pkg="tf" type="static_transform_publisher" name="ugr_axis0" args="-0.36 0.34 0 0 0 0 1 ugr/car_base_link ugr/car_base_link/axis0 10" />
        <node pkg="tf" type="static_transform_publisher" name="ugr_axis1" args="-0.36 -0.34 0 0 0 0 1 ugr/car_base_link ugr/car_base_link/axis1 10" />

        <node pkg="tf" type="static_transform_publisher" name="ugr_imu0" args="-0.26 0 0.35 1 0 0 0 ugr/car_base_link ugr/car_base_link/imu0 10" />
        <node pkg="tf" type="static_transform_publisher" name="ugr_imu1" args="-0.26 0 0.37 1 0 0 0 ugr/car_base_link ugr/car_base_link/imu1 10" />

    </group>

    <group if="$(eval arg('car') == 'pegasus_v2')">

        <node pkg="tf" type="static_transform_publisher" name="ugr_axis0" args="-0.36 0.34 0 0 0 0 1 ugr/car_base_link ugr/car_base_link/axis0 10" />
        <node pkg="tf" type="static_transform_publisher" name="ugr_axis1" args="-0.36 -0.34 0 0 0 0 1 ugr/car_base_link ugr/car_base_link/axis1 10" />

        <node pkg="tf" type="static_transform_publisher" name="ugr_imu0" args="-0.26 0 0.35 1 0 0 0 ugr/car_base_link ugr/car_base_link/imu0 10" />
        <node pkg="tf" type="static_transform_publisher" name="ugr_imu1" args="-0.26 0 0.37 0 0 0 1 ugr/car_base_link ugr/car_base_link/imu1 10" />

    </group>


    <node pkg="can_node" type="can_converter.py" name="can_converter" output="screen">
        <param name="res_send_interval" value="0.1" />
        
        <param name="odrive_dbc" value="$(find can_node)/odrive.dbc" />

        <param name="imu0/frame" value="ugr/car_base_link/imu0" />
        <param name="imu1/frame" value="ugr/car_base_link/imu1" />

        <remap from="/input/can" to="/ugr/car/can/rx" />
        <remap from="/output/can" to="/ugr/car/can/tx" />

        <!-- Encoder 0: right -->
        <!-- Encoder 1: left -->
        <remap from="/output/vel0" to="/ugr/car/sensors/encoder0" />
        <remap from="/output/vel1" to="/ugr/car/sensors/encoder1" />

        <!-- IMU0 : front -->
        <!-- IMU1 : back -->
        <remap from="/output/imu0/pitch_roll" to="/ugr/car/sensors/imu0/pitch_roll" />
        <remap from="/output/imu1/pitch_roll" to="/ugr/car/sensors/imu1/pitch_roll" />
        <remap from="/output/imu0/angular_rate" to="/ugr/car/sensors/imu0/angular_rate" />
        <remap from="/output/imu1/angular_rate" to="/ugr/car/sensors/imu1/angular_rate" />
        <remap from="/output/imu0/acc" to="/ugr/car/sensors/imu0/acc" />
        <remap from="/output/imu1/acc" to="/ugr/car/sensors/imu1/acc" />
    </node>

    <node pkg="can_node" type="can_driver.py" name="can_driver" output="screen">
        
        <param name="can_interface" value="/dev/ttyACM0" />

        <remap from="/input/can" to="/ugr/car/can/tx" />
        <remap from="/output/can" to="/ugr/car/can/rx" />
    </node>
</launch>
